connector_test.go

```diff
@@ -123,37 +116,59 @@ var volumeConfig = `
 }
 `
 
-func TestSimpleVolume(t *testing.T) {
-	logger := log.NewTestLogger(t)
+func bindMountHelper(t *testing.T, options string) {
 	fileContent, err := os.ReadFile("./tests/volume/test_file.txt")
 	assert.NoError(t, err)
 
-	connector, _ := getConnector(t, volumeConfig)
-	cwd, err := os.Getwd()
-	assert.NoError(t, err)
-	// disable selinux on the test folder in order to make the file readable from within the container
-	cmd := exec.Command("chcon", "-Rt", "svirt_sandbox_file_t", fmt.Sprintf("%s/tests/volume", cwd)) //nolint:gosec
-	err = cmd.Run()
-	if err != nil {
-		logger.Warningf("failed to set SELinux permissions on folder, chcon error: %s, this may cause test failure if SELinux is enabled.", err.Error())
+	connector, _ := getConnector(t, fmt.Sprintf(volumeConfig, options))
+
+	if tests.IsRunningOnLinux() && options == "" {
+		// On Linux, when SELinux is enabled, then bind mounts without
+		// relabeling options will fail.  So, to test this case, disable
+		// SELinux on the test folder in order to make the file readable
+		// from within the container.
+		cmd := exec.Command("chcon", "-Rt", "svirt_sandbox_file_t", "./tests/volume")
```

Comment:
OK, this is actually interesting. Yeah, it did this before, and we don't want to do this with selinux labeling options. But ... if the real deployment engine isn't doing this, maybe the tests ought to not be doing it, too, and validating the lack of a label as a failure case?

(Also, isn't this technically wrong anyway, as you're running cases with non-selinux options, and the `&& options == ""` will inhibit the fixup for `:ro`, which ought to require it...)

Reply:
Oh, Dave...every time I make the test tighter you get more right-er.

Now that I have the test creating its own file instead of reusing the same file for all the cases, the `:ro` case is, in fact, failing as you said it should.  🤣

I'm seriously starting to wonder how this test _should_ be designed!

Reply:
So it's increasingly clear that this was a badly busted test from the beginning, and in a way I'm almost nearly slightly sorry I pushed you to start peeling the onion ... 😦

Reply:
Yes...I think the test was good intentioned but not well-executed.  The problem now is, I'm not sure what the requirements actually are or should be.  (Just as we think there is no point to testing `:Z` and `:z` on non-Linux systems, is there any point, when SELinux is enabled, to testing mounts _without_ them??  And, what does this say about the portability message...that we're portable unless you use platform specific features like SELinux options or even bind-mounts?)

I'd love to chat about this in real-time....